src/editors/starrating.js

```diff
@@ -22,7 +30,7 @@ JSONEditor.defaults.editors.starrating = JSONEditor.defaults.editors.string.exte
 
     for(var i = this.enum_values.length-1; i>-1; i--) {
 
-      var id = this.key + '-' + i;
+      var id = this.formname + (i+1);
```

Comment:
This line here was the bug that crash the tests. In the old rating we had 

`[id="root[rating]5"]
[id="root[rating]4"]
[id="root[rating]3"]
[id="root[rating]2"]
[id="root[rating]1"]`

The test fail with the new rating becaus of the index start from 0

`[id="root[rating]4"]
[id="root[rating]3"]
[id="root[rating]2"]
[id="root[rating]1"]
[id="root[rating]0"]`